core/arch/arm/plat-zynqmp/platform_config.h

```diff
@@ -117,6 +119,18 @@
 #define DEVICE2_SIZE		CORE_MMU_DEVICE_SIZE
 #define DEVICE2_TYPE		MEM_AREA_IO_SEC
 
+#ifdef CFG_ZYNQMP_BSLOAD_STA
+# define DEVICE3_PA_BASE	ROUNDDOWN(CSUDMA_BASE, CORE_MMU_DEVICE_SIZE)
```

Comment:
Please use `register_phys_mem()` instead, for instance in `core/arch/arm/plat-zynqmp/sta_bsload.c`.
Same for `DEVICE4_*` below.